packages/next/src/index.ts

```diff
@@ -2575,3 +2584,23 @@ export const prepareCache: PrepareCache = async ({
   debug('Cache file manifest produced');
   return cache;
 };
+
+async function getServerlessPages(params: {
+  pagesDir: string;
+  entryPath: string;
+  outputDirectory: string;
+}) {
+  const [pages, middlewareManifest] = await Promise.all([
+    glob('**/!(_middleware).js', params.pagesDir),
+    getMiddlewareManifest(params.entryPath, params.outputDirectory),
+  ]);
+
+  // Edge Functions are not considered as Serverless Functions
+  for (const edgeFunctionFile of Object.keys(
+    middlewareManifest?.functions ?? {}
+  )) {
+    delete pages[edgeFunctionFile.slice(1) + '.js'];
```

Comment:
Are we planning to add an error when `runtime: 'edge'` is used with a `getStaticProps` page? Currently it isn't compatible with the `Prerender` type as it expects the `lambda` output so will break for that case.

Reply:
That's a good question, but right now with https://github.com/vercel/next.js/pull/37481 it's only going to be api endpoints and not pages, so `Prerender` is not expected to be emitted for this output.

cc @shuding that mentioned some work related to pages 👀

Reply:
Might be good to add an invariant here to ensure we only delete API pages for now to be safe then

Reply:
Yeah I'm going to open a PR for that

Reply:
Soon we will have to remove this constraint as we're refactoring the edge SSR implementation to use functions here :)